prov/sm2/src/sm2_progress.c

```diff
@@ -122,6 +122,54 @@ static int sm2_progress_inject(struct sm2_xfer_entry *xfer_entry,
 	return FI_SUCCESS;
 }
 
+static int sm2_progress_ipc(struct sm2_xfer_entry *xfer_entry,
+			    struct ofi_mr **mr, struct iovec *iov,
+			    size_t iov_count, size_t *total_len,
+			    struct sm2_ep *ep)
+{
+	void *ptr;
+	ssize_t hmem_copy_ret;
+	int ret, err;
+	struct sm2_domain *domain;
+	struct ofi_mr_entry *mr_entry;
+	struct ipc_info *ipc_info = (struct ipc_info *) xfer_entry->user_data;
+
+	domain = container_of(ep->util_ep.domain, struct sm2_domain,
+			      util_domain);
+
+	ret = ofi_ipc_cache_search(domain->ipc_cache,
+				   xfer_entry->hdr.sender_gid, ipc_info,
+				   &mr_entry);
+
```

Comment:
Extra space here if you feel like removing it. If you're not going to update the patch set, don't worry about it.